runtime/src/commonMain/kotlin/dev/zacsweers/metro/internal/BaseDoubleCheck.kt

```diff
@@ -60,7 +60,7 @@ public abstract class BaseDoubleCheck<T>(provider: Provider<T>) :
       }
     }
 
-  override fun isInitialized(): Boolean = _value === UNINITIALIZED
+  override fun isInitialized(): Boolean = _value !== UNINITIALIZED
```

Comment:
This is... weird as I'm pretty sure this was autoconverted from Java 🫣. There's some tests too

Reply:
Errr wait no this would be copied from the stdlib lazy

Reply:
Yeah Dagger lazy doesn't have isInitialized fun, do you expect more test cases here?

Reply:
Yeah if you don't mind adding one that would be good

Reply:
I don't mind, just need to know what scenario else to cover 🤔 I added one in this PR already, or do you expect tests for the BaseDoubleCheck level?

Reply:
Oops missed that on mobile 🫣